cmd/gcp-controller-manager/kubelet_readonly_csr_approver.go

```diff
@@ -252,22 +254,57 @@ func validatePodAnnotation(request kubeletReadonlyCSRRequest) kubeletReadonlyCSR
 		klog.Errorf("csr does not have an attached pod name")
 		return kubeletReadonlyCSRResponse{
 			result:  false,
-			err:     fmt.Errorf("csr does not have an attached pod name"),
+			err:     nil,
 			message: "csr does not have pod name attached",
 		}
 	}
 
+	podUIDs, ok := request.csr.Spec.Extra[podUIDKey]
+	if !ok {
+		klog.Errorf("csr does not have an attached pod UID")
+		return kubeletReadonlyCSRResponse{
+			result:  false,
+			err:     nil,
+			message: "csr does not have pod UID attached",
+		}
+	}
+
+	if len(podNames) != len(podUIDs) {
```

Comment:
naive question, but are UIDs and names supposed to have the same length?

Reply:
Yeah, it supposed to, just in case, I check here if any bad request